test/chain-test.js

```diff
@@ -730,6 +731,289 @@ describe('Chain', function() {
     });
   });
 
+  describe('Covenant limits', function() {
+    before(async () => {
+      await chain.reset(0);
+      wallet = new MemWallet({ network });
+
+      wallet.getNameStatus = async (nameHash) => {
+        assert(Buffer.isBuffer(nameHash));
+        const height = chain.height + 1;
+        const state = await chain.getNextState();
+        const hardened = state.hasHardening();
+        return chain.db.getNameStatus(nameHash, height, hardened);
+      };
+    });
+
+    it('should mine 2000 blocks', async () => {
+      miner.addresses.length = 0;
+      miner.addAddress(wallet.getReceive());
+
+      for (let i = 0; i < 2000; i++) {
+        const block = await cpu.mineBlock();
+        assert(block);
+        assert(await chain.add(block));
+      }
+
+      assert.strictEqual(chain.height, 2000);
+    });
+
+    it('should fail to connect too many OPENs', async () => {
+      const job1 = await cpu.createJob();
+
+      for (let i = 0; i < consensus.MAX_BLOCK_OPENS + 1; i++) {
+        const name = `test_${i}`;
+        const open = await wallet.sendOpen(name);
+        const item = BlockEntry.fromTX(open.toTX(), open.view, job1.attempt);
+        job1.attempt.items.push(item);
+      }
+
+      job1.attempt.refresh();
+
+      assert.strictEqual(await mineBlock(job1), 'bad-blk-opens');
+
+      // "zap" memwallet
+      for (const {tx} of job1.attempt.items) {
+        wallet.removeTX(tx);
+      }
+    });
+
+    it('should connect max number of OPENs', async () => {
+      // We do this 3 times to open enough names for the following tests
+      for (let x = 0; x < 3; x++) {
```

Comment:
Maybe rename `x` to `b` or `blockIndex` ?

It becomes harder to understand what `i` and `x` stand for below, unless you start from the beginning.

Reply:
ok